app/src/main/java/io/bankingofthings/iot/interactors/CheckAndExecuteOfflineActionsWorker.kt

```diff
@@ -0,0 +1,36 @@
+package io.bankingofthings.iot.interactors
+
+import io.bankingofthings.iot.storage.SpHelper
+import io.bankingofthings.iot.utils.ActionUtil
+import io.reactivex.Completable
+import io.reactivex.CompletableSource
+import io.reactivex.Single
+
+class CheckAndExecuteOfflineActionsWorker(
+    private val spHelper: SpHelper,
+    private val triggerActionWorker: TriggerActionWorker
+) {
+    /**
+     * Recursive, trigger one offline action at CORE and then check unitl none is left.
+     */
+    fun execute(): Completable {
```

Comment:
this is weird. You have method, that returns Completable and yet you creating a Single.
I think it's better just use `Completable.create { }` here

Reply:
Completable.create will give unnecessary responsibility with the emitter handeling. Going from Single stream to Completable stream is fine. Starting the stream with a single and processing it.